source/MaterialXTest/ShaderGen.cpp

```diff
@@ -145,27 +145,28 @@ void createLightRig(mx::DocumentPtr doc, mx::LightHandler& lightHandler, mx::HwS
     mx::LightSourcePtr dirLight = lightHandler.createLightSource(dirLightNodeDef);
     dirLight->setParameter("direction", mx::Vector3(0, 0, -1));
     dirLight->setParameter("color", mx::Color3(1, 1, 1));
-    dirLight->setParameter("intensity", 0.4f);
-
+    dirLight->setParameter("intensity", 0.2f);
+    
     mx::LightSourcePtr pointLight = lightHandler.createLightSource(pointLightNodeDef);
-    pointLight->setParameter("position", mx::Vector3(200, 200, 50));
-    pointLight->setParameter("color", mx::Color3(1, 1, 0));
-    pointLight->setParameter("intensity", 10000.0f);
-    pointLight->setParameter("decayRate", 2.0f);
+    pointLight->setParameter("position", mx::Vector3(-2, -2, 2));
+    pointLight->setParameter("color", mx::Color3(0, 0.0, 1));
+    pointLight->setParameter("intensity", 10.0f);
+    pointLight->setParameter("decayRate", 3.0f);
 
     mx::LightSourcePtr spotLight = lightHandler.createLightSource(spotLightNodeDef);
-    spotLight->setParameter("position", mx::Vector3(350, 350, 100));
-    spotLight->setParameter("direction", mx::Vector3(0, 0, -1));
+    spotLight->setParameter("position", mx::Vector3(3, 3, 3));
+    spotLight->setParameter("direction", mx::Vector3(-3, -3, -3));
```

Comment:
Using a non-normalized direction. This reminds me we should make sure the directions are normalised somewhere, I don't think we do that anywhere currently. I guess it's better to do it on application side than having to normalise every vector on shader side, so maybe this is the right place to do it? If we add math functions to the VectorN class, as discussed earlier, this would be easy to do here.

Reply:
I can probably do this in the lighthandler. Will need to find out what's happening with VectorN, MatrixN.

Reply:
I'll take a look into adding at least normalize to VectorN. As to where, it's hard to tell where is best. Generally, I've been paranoid for hw shaders and embedd normalization into the shader. I think we need to do this anyways for normal transforms in the vertex shader (which does not occur) -- it just happens to work for now since there is no scaling factor in the matrix transforms used. For others it makes sense to do it on the app side. I can add in normalization code to the LightHandler binding.